mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/dsl/syntax/XmlDslSyntaxResolver.java

```diff
@@ -370,28 +364,13 @@ private void resolveGroupsDsl(ParameterizedModel component, DslElementSyntaxBuil
         .forEach(parameter -> dsl.containing(parameter.getName(), resolve(parameter)));
   }
 
-  private void addAttributeName(DslElementSyntaxBuilder builder, ParameterModel parameter,
-                                boolean isContent, ParameterDslConfiguration dslModel) {
-
-    if (supportsAttributeDeclaration(parameter, isContent, dslModel)) {
-      builder.withAttributeName(parameter.getName());
-    } else {
-      // For Content parameters, we don't allow the attribute to be set
-      builder.supportsAttributeDeclaration(false);
-    }
-  }
-
-  private boolean supportsAttributeDeclaration(ParameterModel parameter, boolean isContent, ParameterDslConfiguration dslModel) {
-    return !isContent && (dslModel.allowsReferences() || !NOT_SUPPORTED.equals(parameter.getExpressionSupport()));
-  }
-
   private MetadataTypeVisitor getArrayItemTypeVisitor(final DslElementSyntaxBuilder listBuilder, final String parameterName,
                                                       final String namespace, final String namespaceUri, boolean asItem) {
     return new MetadataTypeVisitor() {
 
       @Override
       public void visitObject(ObjectType objectType) {
-        if (isMap(objectType)) {
+        if (isMap(objectType) && !isInfrastructure(objectType)) {
```

Comment:
isn't it better if `isMap()` checks for the type being infrastructure?? The reality is that the infrastructure parameters are not real maps. We're using map just because we cannot use the real type and Map is a nice default. 

But applied like this, we have a potential bug each time somebody uses isMap() without the second clause.